android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.kt

```diff
@@ -0,0 +1,326 @@
+      )
+    }
+  }
+
+  private fun requestOverlayPermission(context: Context) {
+    // From the unexposed DebugOverlayController static helper
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      // Get permission to show debug overlay in dev builds.
+      if (!Settings.canDrawOverlays(context)) {
+        val intent = Intent(
+          Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+          Uri.parse("package:" + context.packageName)
+        ).apply {
+          flags = Intent.FLAG_ACTIVITY_NEW_TASK
+        }
+        FLog.w(
+          ReactConstants.TAG,
+          "Overlay permissions needs to be granted in order for React Native apps to run in development mode"
+        )
+        if (intent.resolveActivity(context.packageManager) != null) {
+          context.startActivity(intent)
+        }
+      }
+    }
+  }
+
+  private fun togglePerformanceMonitor() {
+    val currentActivity = Exponent.instance.currentActivity
+    if (currentActivity is ReactNativeActivity) {
+      val devSettings = currentActivity.devSupportManager.callRecursive("getDevSettings")
+      if (devSettings != null) {
+        val isFpsDebugEnabled = devSettings.call("isFpsDebugEnabled") as Boolean
+        if (!isFpsDebugEnabled) {
+          // Request overlay permission if needed when "Show Perf Monitor" option is selected
+          requestOverlayPermission(currentActivity)
+        }
+        devSettings.call("setFpsDebugEnabled", !isFpsDebugEnabled)
+      }
+    } else {
+      FLog.e(
+        ReactConstants.TAG,
+        "Unable to toggle the performance monitor because the current activity could not be found."
+      )
+    }
+  }
+
+  private fun toggleRemoteJSDebugging() {
+    val currentActivity = Exponent.instance.currentActivity
+    if (currentActivity is ReactNativeActivity) {
+      val devSettings = currentActivity.devSupportManager.callRecursive("getDevSettings")
+      if (devSettings != null) {
+        val isRemoteJSDebugEnabled = devSettings.call("isRemoteJSDebugEnabled") as Boolean
+        devSettings.call("setRemoteJSDebugEnabled", !isRemoteJSDebugEnabled)
+      }
+    } else {
+      FLog.e(
+        ReactConstants.TAG,
+        "Unable to toggle remote JS debugging because the current activity could not be found."
+      )
+    }
+  }
+
+  private fun createPackagerCommandHelpers(): Map<String, RequestHandler> {
+    // Attach listeners to the bundler's dev server web socket connection.
+    // This enables tools to automatically reload the client remotely (i.e. in expo-cli).
+    val packagerCommandHandlers = mutableMapOf<String, RequestHandler>()
+
+    // Enable a lot of tools under the same command namespace
+    packagerCommandHandlers["sendDevCommand"] = object : NotificationOnlyHandler() {
+      override fun onNotification(params: Any?) {
+        if (params != null && params is JSONObject) {
+          val name = if (params.has("name")) {
```

Comment:
I left a previous comment on another PR, but essentially optString doesn't return null when the key doesn't exist. It returns empty string instead, which in this case is fine since the case isn't handled in the `when`, but I like explicit null better. I'm planning to add a JSONObject extension method that does this better.

Reply:
Something like jsonObject.nullableString() does seem better